### ISSUE TITLE
GCC compilation error

### ISSUE THREAD

**Issue by HeyDon-Lee** [go-vgo/robotgo#233]
* Code:
```go
package main

import "github.com/go-vgo/robotgo"

func main() {
	for {
		_, y := robotgo.GetMousePos()
		println(y)
		if y < 35 || y > 1037 {
			robotgo.MoveMouse(960, 540)
		}
	}
}
```

* Error:
```
c:\go\pkg\tool\windows_amd64\link.exe: running gcc failed: exit status 1
C:/Program Files/mingw-w64/x86_64-8.1.0-posix-seh-rt_v6-rev0/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: i386 architecture of input file `C:\Users\12526\AppData\Local\Temp\go-link-025442471\000000.o' is incompatible with i386:x86-64 output
collect2.exe: error: ld returned 1 exit status
```

**Comment by vcaesar** [comment]
Closing is due to can't reproduce. Check the gcc environment and version.

**Comment by 502647092** [comment]
OS: Windows 10
robotgo: master

```
$ gcc -v
Using built-in specs.
COLLECT_GCC=C:\Program Files\mingw-w64\x86_64-8.1.0-posix-seh-rt_v6-rev0\mingw64\bin\gcc.exe
COLLECT_LTO_WRAPPER=C:/Program\ Files/mingw-w64/x86_64-8.1.0-posix-seh-rt_v6-rev0/mingw64/bin/../libexec/gcc/x86_64-w64-mingw32/8.1.0/lto-wrapper.exe
Target: x86_64-w64-mingw32
Configured with: ../../../src/gcc-8.1.0/configure --host=x86_64-w64-mingw32 --build=x86_64-w64-mingw32 --target=x86_64-w64-mingw32 --prefix=/mingw64 --with-sysroot=/c/mingw810/x86_64-810-posix-seh-rt_v6-rev0/mingw64 --enable-shared --enable-static --disable-multilib --enable-languages=c,c++,fortran,lto --enable-libstdcxx-time=yes --enable-threads=posix --enable-libgomp --enable-libatomic --enable-lto --enable-graphite --enable-checking=release --enable-fully-dynamic-string --enable-version-specific-runtime-libs --disable-libstdcxx-pch --disable-libstdcxx-debug --enable-bootstrap --disable-rpath --disable-win32-registry --disable-nls --disable-werror --disable-symvers --with-gnu-as --with-gnu-ld --with-arch=nocona --with-tune=core2 --with-libiconv --with-system-zlib --with-gmp=/c/mingw810/prerequisites/x86_64-w64-mingw32-static --with-mpfr=/c/mingw810/prerequisites/x86_64-w64-mingw32-static --with-mpc=/c/mingw810/prerequisites/x86_64-w64-mingw32-static --with-isl=/c/mingw810/prerequisites/x86_64-w64-mingw32-static --with-pkgversion='x86_64-posix-seh-rev0, Built by MinGW-W64project' --with-bugurl=https://sourceforge.net/projects/mingw-w64 CFLAGS='-O2 -pipe -fno-ident -I/c/mingw810/x86_64-810-posix-seh-rt_v6-rev0/mingw64/opt/include -I/c/mingw810/prerequisites/x86_64-zlib-static/include -I/c/mingw810/prerequisites/x86_64-w64-mingw32-static/include' CXXFLAGS='-O2 -pipe -fno-ident -I/c/mingw810/x86_64-810-posix-seh-rt_v6-rev0/mingw64/opt/include -I/c/mingw810/prerequisites/x86_64-zlib-static/include -I/c/mingw810/prerequisites/x86_64-w64-mingw32-static/include' CPPFLAGS=' -I/c/mingw810/x86_64-810-posix-seh-rt_v6-rev0/mingw64/opt/include -I/c/mingw810/prerequisites/x86_64-zlib-static/include -I/c/mingw810/prerequisites/x86_64-w64-mingw32-static/include' LDFLAGS='-pipe -fno-ident -L/c/mingw810/x86_64-810-posix-seh-rt_v6-rev0/mingw64/opt/lib -L/c/mingw810/prerequisites/x86_64-zlib-static/lib -L/c/mingw810/prerequisites/x86_64-w64-mingw32-static/lib '
Thread model: posix
gcc version 8.1.0 (x86_64-posix-seh-rev0, Built by MinGW-W64 project)
```

ERROR
```
$ CGO_ENABLED=1 go build -ldflags="-s -w -H windowsgui"
go: finding github.com/shirou/w32 latest
# yumc.pw/cloud/sixi-pc-plugin
C:\Go\pkg\tool\windows_amd64\link.exe: running gcc failed: exit status 1
C:/Program Files/mingw-w64/x86_64-8.1.0-posix-seh-rt_v6-rev0/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: i386 architecture of input file `C:\Users\jtb\AppData\Local\Temp\go-link-525021911\000000.o' is incompatible with i386:x86-64 output
collect2.exe: error: ld returned 1 exit status
```

**Comment by 100000000resU** [comment]
C:\Windows\system32>go get github.com/go-vgo/robotgo
# github.com/go-vgo/robotgo
C:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lz
collect2.exe: error: ld returned 1 exit status

I am having this issue if anyone has solved it. I have tried reinstalling gcc using the standard tdm64-gcc-9.2.0 from jmeubank.github.io already and a few other random things with no luck. Any help would be appreciated.

**Comment by xiaofsu** [comment]
> C:\Windows\system32>go get github.com/go-vgo/robotgo
> 
> # github.com/go-vgo/robotgo
> C:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lz
> collect2.exe: error: ld returned 1 exit status
> 
> I am having this issue if anyone has solved it. I have tried reinstalling gcc using the standard tdm64-gcc-9.2.0 from jmeubank.github.io already and a few other random things with no luck. Any help would be appreciated.

How is it solved?

**Comment by vcaesar** [comment]
Closed because of duplicates, #100. All code tests and C dependencies use MinGw-64, TDM-GCC has not been tested.
But you can compile lpng and lzib yourself, and you can download https://sourceforge.net/projects/mingw-w64/files/ x86_64-posix.

**Comment by vcaesar** [comment]
All code tests and C dependencies use MinGw-64, TDM-GCC has not been tested.
But you can compile `lpng and lzib` yourself, and you can download https://sourceforge.net/projects/mingw-w64/files/ x86_64-posix.
